custom_components/storj/backup.py

```diff
@@ -138,7 +142,9 @@ async def async_delete_backup(
 
             if backup:
                 await self._client.async_delete_backup(backup)
+                return
         except (UplinkError, HomeAssistantError, TimeoutError) as err:
             raise BackupAgentError(
                 f"Failed to delete backup {backup_id}: {err}"
             ) from err
+        raise BackupNotFound(f"Backup {backup_id} not found")
```

Comment:
I don't think this is needed, `async_get_backup` will raise if the backup does not exist
```suggestion
```